src/main/java/com/redhat/red/build/koji/http/httpclient4/HC4SyncObjectClient.java

```diff
@@ -40,28 +42,51 @@
 import java.net.MalformedURLException;
 import java.util.Arrays;
 
+import static com.codahale.metrics.MetricRegistry.name;
+
 public class HC4SyncObjectClient
 {
     private final HttpFactory httpFactory;
 
     private final SiteConfig siteConfig;
 
+    private final MetricRegistry metricRegistry;
+
     private final String[] extraPath;
 
-    public HC4SyncObjectClient( final HttpFactory httpFactory, final SiteConfig siteConfig, String... extraPath )
+    public HC4SyncObjectClient( final HttpFactory httpFactory, final SiteConfig siteConfig,
+                                final MetricRegistry metricRegistry, String... extraPath )
     {
         this.httpFactory = httpFactory;
         this.siteConfig = siteConfig;
+        this.metricRegistry = metricRegistry;
         this.extraPath = extraPath;
     }
 
-    public <T> T call( final Object request, final Class<T> responseType ) throws XmlRpcException
+    public <T> T call( final Object request, final Class<T> responseType, final UrlBuilder urlBuilder,
+                       final RequestModifier requestModifier ) throws XmlRpcException
     {
-        return call( request, responseType, null, null );
+        if ( metricRegistry == null )
+        {
+            return doCall( request, responseType, urlBuilder, requestModifier );
+        }
+
+        // Apply metric
+
+        final Timer timer = metricRegistry.timer( name( getClass(), "call" ) );
```

Comment:
Can we incorporate the type of call here, so we can split the xml-rpc methods? I'm not sure how to achieve that from this class, but if we can separate by method that would be much more useful. Otherwise, we're not going to get much in the way of useful information out of this timing.